src/index.ts

```diff
@@ -0,0 +1,40 @@
+//1
+interface Interface1 {
+	[key: string]: string | number;
+}
+
+interface Interface2 {
+	[key: number]: (() => void);
+}
+
+interface Interface3 {
+	[key: number]: string; // не зовсiм зрозумiв "...а значення - певного типу." Якого саме?
+}
+
+interface Interface4 {
+	name: string;
+	[key: string]: string | number;	
+}
+
+
+
+// ---5---
+type RandomObject = {
+	[key: string]: string | number;
+}
+interface MyInterface {
+	[key: string]: string | number | RandomObject;
+}
+interface MyInterface {
+	property1: string;
+	property2: number;
+	property3: RandomObject
```

Comment:
I should admit that it's an interesting interpretation; however, the task mentioned extending one interface from another, not just a single field. It will look like:
```
interface IParent {
// some fields
}
interface IChild extends IParent {
// some additiona fields
}
```

Reply:
ok